File: fixtures/typescript/arrow_function_duplicate/left.ts
export const summarizeOrders = (orders: Order[]): Summary => {
  const paid = orders.filter((order) => order.paid);
  const totals = paid.map((order) => order.amount);
  const count = totals.length;
  const sum = totals.reduce((left, right) => left + right, 0);
  return { count, sum };
};

File: fixtures/rust/renamed_function_duplicate/left.rs
pub fn invoice_summary(orders: &[Order]) -> Summary {
    let paid = orders.iter().filter(|order| order.paid).collect::<Vec<_>>();
    let domestic = paid.iter().filter(|order| order.country == "US").collect::<Vec<_>>();
    let total = domestic.iter().map(|order| order.amount).sum::<i64>();
    let flagged = domestic.iter().filter(|order| order.flagged).count();
    Summary { total, flagged }
}
